### PR TITLE
Add DKIM tests

### PR DESCRIPTION
**Pull request by saleel**
## Description

- Add reason for DKIM failure
- Add tests for DKIM signature check (helpers package)
- Refactor ARC revert handling

## Type of Change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

## Checklist:

- [ ] I have discussed with the team prior to submitting this PR
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] My changes generate no new warnings
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: packages/helpers/src/dkim/index.ts [hunk 1]
```diff
@@ -87,39 +103,15 @@ async function tryVerifyDKIM(email: Buffer | string, domain: string = "") {
...
-
-  return headerValue;
-}
-
-function setHeaderValue(email: string, header: string, value: string) {
-  return email.replace(getHeaderValue(email, header), value);
-}
-
-async function revertForwarderChanges(email: string) {
-  // Google sets their own Message-ID and put the original one in X-Google-Original-Message-ID when forwarding
```
Comment (Divide-By-0) [inline packages/helpers/src/dkim/index.ts:103]: I'm worried the new code doesn't have coverage on this due to lack of tests, are you sure your changes work for this DKIM signature and message ID?

File: packages/helpers/src/dkim/index.ts [hunk 2]
```diff
@@ -22,20 +28,28 @@ export async function verifyDKIMSignature(
 ): Promise<DKIMVerificationResult> {
   let dkimResult = await tryVerifyDKIM(email, domain);
 
-  if (dkimResult.status.result !== "pass" && tryRevertARCChanges) {
-    console.info("DKIM verification failed. Trying to verify after reverting forwarder changes...");
-
-    const modified = await revertForwarderChanges(email.toString());
+  // If DKIM verification fails, revert common modifications made by ARC and try again.
+  if (dkimResult.status.comment === "bad signature" && tryRevertARCChanges) {
```
Comment (Divide-By-0) [inline packages/helpers/src/dkim/index.ts:32]: before we do this, we should add the zkp2p code where they try to process the email via doing the tabs -> spaces replacement as well as removing a [LABEL] tag from an email. i think we shuold try as hard as possible to get a dkim before moving to arc, and output that dkim failed and we are moving to arc before doing modifications and claiming dkim passes.

**Comment by Divide-By-0** [comment]
Might also be good to include the modifications in https://github.com/zkemail/zk-email-verify/issues/162 here, maybe even https://github.com/zkemail/zk-email-verify/issues/165 (the assigned guy isn't doing it). The zkp2p fixes are here:

see: https://github.com/zkp2p/zk-p2p/blob/develop/client/src/components/ProofGen/validation/hdfc.tsx#L85
and: https://github.com/zkp2p/zk-p2p/blob/develop/client/src/components/ProofGen/validation/venmo.tsx#L62

File: packages/helpers/src/dkim/index.ts (hunk 2 above)
Comment (saleel) [reply packages/helpers/src/dkim/index.ts:32]: > tabs -> spaces replacement as well as removing a [LABEL] tag from an email
Do you know where this is implemented? I think those changes are also part of ARC? We should definitely be adding other ARC changes.

> output that dkim failed and we are moving to arc
The `revertCommonARCModifications` will print a log when modifications are done, or if no changes are done.

File: packages/helpers/src/dkim/index.ts (hunk 1 above)
Comment (saleel) [reply packages/helpers/src/dkim/index.ts:103]: Unfortunately I dont have a non-sensitive email that contains ARC changes which can be pushed to git. I used the one I have with existing tests and they are passing (after successful modifications)

File: packages/helpers/src/dkim/index.ts (hunk 2 above)
Comment (Divide-By-0) [reply packages/helpers/src/dkim/index.ts:32]: > > tabs -> spaces replacement as well as removing a [LABEL] tag from an email
> 
> Do you know where this is implemented? I think those changes are also part of ARC? We should definitely be adding other ARC changes.
> 
> > output that dkim failed and we are moving to arc
> 
> The `revertCommonARCModifications` will print a log when modifications are done, or if no changes are done.

Check out the zkp2p fixes -- basically like if you revert those changes, then you can get DKIM to pass, not just ARC. I think they noticed this on some Venmo emails and diffed them.

Comment (saleel) [reply packages/helpers/src/dkim/index.ts:32]: We are doing the same approach - right now only Message-Id is reverted, but can add removing labels.

https://github.com/zkp2p/zk-p2p/blob/develop/client/src/components/ProofGen/validation/hdfc.tsx#L85 seems to be very specific to hdfc pasted emails.

I think we can merge this and address 162, 165 and other sanitizations in another PR.

**Pull request merged**